mmv1/templates/terraform/pre_delete/dialogflowcx_agent.go.tmpl

```diff
@@ -0,0 +1,53 @@
+s := d.Get("gen_app_builder_settings")
+log.Printf("[DEBUG] gen_app_builder_settings: %v", s)
+settings, ok := s.([]interface{})
+if !ok {
+	return fmt.Errorf("Error converting gen_app_builder_settings %s to  []interface{}", s)
+}
+if len(settings) == 0 {
+    // no engine to delete
+    return nil
+}
+
+// get the engine ID
+engineIntf, ok := settings[0].(map[string]interface{})["engine"]
+if !ok {
+    return fmt.Errorf("Expected key 'engine' in map %+v", settings[0])
+}
+engineID, ok := engineIntf.(string)
+if !ok {
+	return fmt.Errorf("Can convert engine ID %s to string", engineIntf)
+}
+
+// check the engine exist
+baseUrl, err := tpgresource.ReplaceVars(d, config, "{{"{{"}}DiscoveryEngineBasePath{{"}}"}}")
+if err != nil {
+	return err
+}
+engineUrl := baseUrl + engineID
+_, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+	Config:    config,
+	Method:    "GET",
+	Project:   project,
+	RawURL:    engineUrl,
+	UserAgent: userAgent,
+})
+if err != nil {
+	log.Printf("[DEBUG] engine %s doesn't exist. No need to delete", engineID)
+	return nil
```

Comment:
We shouldn't return nil here, correct? If the engine doesn't exist we still want to delete the agent later on